parsons/rockthevote/rtv.py

```diff
@@ -0,0 +1,211 @@
+import datetime
+import logging
+import petl
+import re
+import requests
+import time
+
+from parsons import Table
+from parsons.utilities import check_env
+from parsons.utilities.api_connector import APIConnector
+
+logger = logging.getLogger(__name__)
+
+DATETIME_FORMAT = '%Y-%m-%d %H:%M:%S UTC'
+"""Datetime format for sending date's to the API."""
+
+REQUEST_HEADERS = {
+    # For some reason, RTV's firewall REALLY doesn't like the
+    # user-agent string that Python's request library gives by default,
+    # though it seems fine with the curl user agent
+    # For more info on user agents, see:
+    # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent
+    'user-agent': 'curl/7.54.0'
+}
+"""Standard request header for sending requests to the API."""
+
+STATUS_URL_PARSE_REGEX = re.compile(r'(\d+)$')
+"""Regex for parsing the report ID from the status URL."""
+
+
+class RTVFailure(Exception):
+    """Exception raised when there is an error with the connector."""
```

Comment:
Noticed this pattern in Sky's code too. Would it make sense to move towards this as a standard?

Reply:
Yeah. I think it's a good idea to raise custom exceptions. An outstanding question is how we handle inheritance: do we have some common `ParsonsException` or do we have our custom exceptions inherit from corollary standard Python exceptions (e.g. `ValueError`). @schuyler1d do you have thoughts?